lib/iris/analysis/cartography.py

```diff
@@ -713,3 +713,311 @@ def project(cube, target_proj, nx=None, ny=None):
+    ds_dy = np.sqrt(dlat_dy * dlat_dy + dlon_dy * dlon_dy * lat_factor)
+    return ds_dx, ds_dy
+
+
+def _transform_distance_vectors(src_crs, x, y, u_dist, v_dist, tgt_crs):
+    """
+    Transform distance vectors from one coordinate reference system to
+    another, preserving magnitude and physical direction.
+
+    Args:
+
+    * src_crs, tgt_crs (`cartopy.crs.Projection`):
+        The source and target coordinate reference systems.
+    * x, y (array):
+        Locations of each vector defined in 'src_crs'.
+    * u_dist, v_dist (array):
+        Components of each vector along the x and y directions of 'src_crs'
+        at each location.
+
+    Returns:
+        (ut_dist, vt_dist): Tuple of arrays containing the vector components
+        along the x and y directions of 'tgt_crs' at each location.
+
+    """
+    # Get distance scalings for source crs.
+    ds_dx1, ds_dy1 = _crs_distance_differentials(src_crs, x, y)
+    # Get distance scalings for target crs.
+    x2, y2 = _transform_xy(src_crs, x, y, tgt_crs)
+    ds_dx2, ds_dy2 = _crs_distance_differentials(tgt_crs, x2, y2)
+    # Scale input distance vectors --> source-coordinate differentials.
+    u1, v1 = u_dist / ds_dx1, v_dist / ds_dy1
+    # Transform vectors into the target system.
+    dx2_x1, dy2_x1, dx2_y1, dy2_y1 = _inter_crs_differentials(
+        src_crs, x, y, tgt_crs)
+    u2 = dx2_x1 * u1 + dx2_y1 * v1
+    v2 = dy2_x1 * u1 + dy2_y1 * v1
+    # Rescale output coordinate vectors --> target distance vectors.
+    u2_dist, v2_dist = u2 * ds_dx2, v2 * ds_dy2
+
+    return u2_dist, v2_dist
+
+
+def rotate_winds(u_cube, v_cube, target_cs):
+    """
+    Transform wind vectors to a different coordinate system.
+
+    The input cubes contain U and V components parallel to the local X and Y
+    directions of the input grid at each point.
+
+    The output cubes contain the same winds, at the same locations, but
+    relative to the grid directions of a different coordinate system.
+    Thus in vector terms, the magnitudes will always be the same, but the
+    angles can be different.
+
+    The outputs retain the original horizontal dimension coordinates, but
+    also have two 2-dimensional auxiliary coordinates containing the X and
+    Y locations in the target coordinate system.
+
+    Args:
+
+    * u_cube
+        An instance of :class:`iris.cube.Cube` that contains the x-component
+        of the vector.
+    * v_cube
+        An instance of :class:`iris.cube.Cube` that contains the y-component
+        of the vector.
+    * target_cs
+        An instance of :class:`iris.coord_systems.CoordSystem` that specifies
+        the new grid directions.
+
+    Returns:
+        A (u', v') tuple of :class:`iris.cube.Cube` instances that are the u
+        and v components in the requested target coordinate system.
+        The units are the same as the inputs.
+
+    ..note ::
+
+        The U and V values are measures of distance, with units such as
```

Comment:
@pp-mo - I find this troublesome as they are measures of speed, not distance. I think I understand your desire for this statement - would replacing distance with speed be ok?